packages/frontend/src/components/wallet/TokenAmount.js

```diff
@@ -1,47 +1,74 @@
-    var parts = amount.split('.');
-    parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',');
-    return parts.join('.');
+    var parts = amount.split(".");
+    parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+    return parts.join(".");
 };
 
 const formatToken = (amount, decimals) => {
-    if (amount === '0') {
+    if (amount === "0") {
         return amount;
     }
 
     let formattedAmount = formatTokenAmount(amount, decimals, FRAC_DIGITS);
 
-    if (formattedAmount === `0.${'0'.repeat(FRAC_DIGITS)}`) {
-        return `< ${!FRAC_DIGITS ? '0' : `0.${'0'.repeat((FRAC_DIGITS || 1) - 1)}1`}`;
+    if (formattedAmount === `0.${"0".repeat(FRAC_DIGITS)}`) {
+        return `< ${
+            !FRAC_DIGITS ? "0" : `0.${"0".repeat((FRAC_DIGITS || 1) - 1)}1`
+        }`;
     }
     return amountWithCommas(removeTrailingZeros(formattedAmount));
 };
 
 const showFullAmount = (amount, decimals, symbol) =>
-    (amount !== '0' && !!amount)
+    amount !== "0" && !!amount
         ? `${formatTokenAmount(amount, decimals, decimals)} ${symbol}`
-        : '';
+        : "";
 
-const TokenAmount = ({ token: { balance, onChainFTMetadata }, withSymbol = false, className, showFiatAmount = true, 'data-test-id': testId }) => (
-    <div className={className} title={showFullAmount(balance, onChainFTMetadata?.decimals, onChainFTMetadata?.symbol)} data-test-id={testId}>
+const TokenAmount = ({
+    token: { balance, onChainFTMetadata, coingeckoMetadata },
+    withSymbol = false,
+    className,
+    showFiatAmount = true,
+    "data-test-id": testId,
+}) => (
+    <div
+        className={className}
+        title={showFullAmount(
+            balance,
+            onChainFTMetadata?.decimals,
+            onChainFTMetadata?.symbol
+        )}
+        data-test-id={testId}
+    >
         <div>
-            {balance
-                ? formatToken(balance, onChainFTMetadata?.decimals)
-                : <span className='dots' />
-            }
-            <span className='currency'>{withSymbol ? ` ${onChainFTMetadata?.symbol}` : null}</span>
+            {balance ? (
+                formatToken(balance, onChainFTMetadata?.decimals)
+            ) : (
+                <span className="dots" />
+            )}
+            <span className="currency">
+                {withSymbol ? ` ${onChainFTMetadata?.symbol}` : null}
+            </span>
         </div>
-        {showFiatAmount &&
-            <div className='fiat-amount'>
-                — USD
+        {showFiatAmount && coingeckoMetadata?.usd ? (
+            <div className="fiat-amount">
+                <BalanceDisplayUSD
+                    amount={balance}
+                    nearTokenFiatValueUSD={coingeckoMetadata?.usd}
+                    decimals={onChainFTMetadata?.decimals}
+                    isNear={true}
+                />
             </div>
-        }
+        ) : (
+            <div className="fiat-amount">— USD</div>
+        )}
```

Comment:
This needs to hide fiat amount if `showFiatAmount` is `false` to retain current behavior

Reply:
Addressed in #2583